doc/icache.rst

```diff
@@ -0,0 +1,130 @@
++-------------------------+-----------+-----------------------------------------------+
+| ``SpecRequest``         | ``1'b0``  | When set, the system will attempt to          |
+|                         |           | speculatively request data from memory in     |
+|                         |           | parallel with the cache lookup. This can give |
+|                         |           | improved performance for workloads which      |
+|                         |           | cache poorly (at the expense of power).       |
++-------------------------+-----------+-----------------------------------------------+
+| ``BranchCache``         | ``1'b0``  | When set, the cache will only allocate the    |
+|                         |           | targets of branches + two subsequent cache    |
+|                         |           | lines. This gives improved performance in     |
+|                         |           | systems with moderate latency by not          |
+|                         |           | polluting the cache with data that can be     |
+|                         |           | prefetched instead.                           |
++-------------------------+-----------+-----------------------------------------------+
+
+RAM Arrangement
+---------------
+
+The data RAMs are arranged as a series of 32 bit banks, the number depending on the cache line width and number of ways.
+Each group of banks forming a single way can be combined into wider RAM instances if required since they are always accessed together.
+
+Indicative RAM sizes for common configurations are given in the table below:
+
++-------------------------+-----------------+-----------------+
+| Cache config            | Tag RAMs        | Data RAMs       |
++=========================+=================+=================+
+| 4kB, 2 way, 64bit line  | 2 x 256 x 21bit | 4 x 256 x 32bit |
++-------------------------+-----------------+-----------------+
+| 4kB, 2 way, 128bit line | 2 x 128 x 21bit | 8 x 128 x 32bit |
++-------------------------+-----------------+-----------------+
+| 4kB, 4 way, 64bit line  | 4 x 128 x 21bit | 8 x 128 x 32bit |
++-------------------------+-----------------+-----------------+
+
+Sub Unit Description
+--------------------
+
+.. figure:: images/icache_block.svg
+   :name: icache_block
+   :align: center
+
+   Instruction Cache Block Diagram
+
+Prefetch Address
+^^^^^^^^^^^^^^^^
+
+The prefetch address is updated to the branch target on every branch.
+This address is then updated in cache-line increments each time a cache lookup is issued to the cache pipeline.
+
+Cache Pipeline
+^^^^^^^^^^^^^^
+
+The cache pipeline consists of two stages, IC0 and IC1.
+
+In IC0, lookup requests are arbitrated against cache allocation requests.
+Lookup requests have fixed priority since they naturally throttle themselves as fill buffer resources run out.
+The arbitrated request is made to the RAMs in IC0.
```

Comment:
Also, I was slightly tripped up by "fixed priority". I guess you mean "highest priority" here (which is fixed because it's always highest!). There's also a comment in ibex_icache.sv with the same terminology.

Reply:
Thanks for the clarification @tomroberts-lowrisc . Maybe the fixed priority could indeed be changed to highest priority as @rswarbrick suggested.

Reply:
:+1: